pkg/hive/cell/module.go

```diff
@@ -56,16 +57,31 @@ func (m *module) logger(log logrus.FieldLogger) logrus.FieldLogger {
 	return log.WithField(logfields.LogSubsys, m.id)
 }
 
-func (m *module) moduleScopedStatusReporter(p Health) HealthReporter {
-	return p.forModule(m.id)
+type reporterHooks struct {
+	rootScope Scope
+}
+
+func (r *reporterHooks) Start(ctx lifecycle.HookContext) error {
+	return nil
+}
+
+func (r *reporterHooks) Stop(ctx lifecycle.HookContext) error {
+	flushAndClose(r.rootScope, "Hive shutting down")
+	return nil
+}
+
+func (m *module) moduleReporterScope(p Health, lc lifecycle.Lifecycle) Scope {
+	rs := rootScope(p.forModule(m.id))
```

Comment:
BTW, I was thinking we might want "fully-qualified module ids" here, e.g. `agent.controlplane.endpoint-manager`. This would be very easy to expose, e.g. have `cell.Module` provide module scoped `type ModuleId string` and `type QualifiedModuleId []ModuleId`. WDYT?

Reply:
https://github.com/cilium/cilium/pull/28512

Reply:
sounds good, we can fix the merge conflicts on whatever PR is merged second.